### ISSUE TITLE
keras model training - total loss discrepancy with multiple outputs

### ISSUE THREAD

**Issue by phemmer** [tensorflow/tensorflow#36006]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Fedora 31
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Source
- TensorFlow version (use command below): v2.1.0-1-ga9af83a149 2.1.0
- Python version: 3.7.5
- Bazel version (if compiling from source): 0.29.1
- GCC/Compiler version (if compiling from source): 8.3.1
- CUDA/cuDNN version: 10.2.89 / 7.6.5.33
- GPU model and memory: Nvidia GeForce GTX 1070 TI 8GB

**Describe the current behavior**
While training a model with multiple outputs via keras, the `loss` and `val_loss` values do not agree with the documented definition. The documentation states:

> The loss value that will be minimized by the model will then be the weighted sum of all individual losses, weighted by the loss_weights coefficients.

However as can be seen by the following screencap of tensorboard, this does not match up:
![image](https://user-images.githubusercontent.com/1826947/72649484-d2c9ea80-394b-11ea-8f44-818ff8b526f3.png)

And the verbose training log output agrees with the tensorboard charts:
```
Epoch 1/30
37/37 [==============================] - 3s 74ms/step - loss: 776.4654 - out_0_loss: 0.0281 - out_1_loss: 0.0189 - out_2_loss: 0.0310 - out_3_loss: 0.0319 - out_4_loss: 0.0525 - out_5_loss: 0.0643 - out_0_mean_squared_error: 0.0281 - out_1_mean_squared_error: 0.0189 - out_2_mean_squared_error: 0.0310 - out_3_mean_squared_error: 0.0319 - out_4_mean_squared_error: 0.0525 - out_5_mean_squared_error: 0.0643 - val_loss: 121.1278 - val_out_0_loss: 0.0116 - val_out_1_loss: 0.0119 - val_out_2_loss: 0.0110 - val_out_3_loss: 0.0136 - val_out_4_loss: 0.0186 - val_out_5_loss: 0.0273 - val_out_0_mean_squared_error: 0.0116 - val_out_1_mean_squared_error: 0.0119 - val_out_2_mean_squared_error: 0.0110 - val_out_3_mean_squared_error: 0.0136 - val_out_4_mean_squared_error: 0.0186 - val_out_5_mean_squared_error: 0.0273

Epoch 2/30
37/37 [==============================] - 2s 45ms/step - loss: 57.5657 - out_0_loss: 0.0251 - out_1_loss: 0.0237 - out_2_loss: 0.0244 - out_3_loss: 0.0311 - out_4_loss: 0.0361 - out_5_loss: 0.0479 - out_0_mean_squared_error: 0.0251 - out_1_mean_squared_error: 0.0237 - out_2_mean_squared_error: 0.0244 - out_3_mean_squared_error: 0.0311 - out_4_mean_squared_error: 0.0361 - out_5_mean_squared_error: 0.0479 - val_loss: 33.7718 - val_out_0_loss: 0.0241 - val_out_1_loss: 0.0222 - val_out_2_loss: 0.0226 - val_out_3_loss: 0.0290 - val_out_4_loss: 0.0349 - val_out_5_loss: 0.0447 - val_out_0_mean_squared_error: 0.0241 - val_out_1_mean_squared_error: 0.0222 - val_out_2_mean_squared_error: 0.0226 - val_out_3_mean_squared_error: 0.0290 - val_out_4_mean_squared_error: 0.0349 - val_out_5_mean_squared_error: 0.0447
```

And since I am using weighted values, the weights are:
```
[1.0, 1.0, 0.8, 0.6, 0.45, 0.3]
```

I'm not using weighted samples or anything. The weights are only defined within `model.compile()`.
The data is fed as a `tf.data.Dataset`, where the shape of the training & validation data is:
```
(TensorSpec(shape=(None, 3, 19546), dtype=tf.float32, name=None), (TensorSpec(shape=(None,), dtype=tf.float32, name=None), TensorSpec(shape=(None,), dtype=tf.float32, name=None), TensorSpec(shape=(None,), dtype=tf.float32, name=None), TensorSpec(shape=(None,), dtype=tf.float32, name=None), TensorSpec(shape=(None,), dtype=tf.float32, name=None), TensorSpec(shape=(None,), dtype=tf.float32, name=None)))
```

**Describe the expected behavior**
For epoch 1:
```
weights = np.array([1.0, 1.0, 0.8, 0.6, 0.45, 0.3])
losses = np.array([0.0281, 0.0189, 0.0310, 0.0319, 0.0525, 0.0643])
val_losses = np.array([0.0116, 0.0119, 0.0110, 0.0136, 0.0186, 0.0273])

loss = np.sum(losses * weights)
0.133855

val_loss = np.sum(val_losses * weights)
0.05702
```

**Code to reproduce the issue**
Have been trying to reproduce with a simple example I can share, but am having difficulty. The little bit of info I can provide is how the model was compiled, and fit:
```
optimizer = tf.keras.optimizers.Adam(learning_rate=0.0005)
model.compile(loss='mean_squared_error', loss_weights=[1.0,1.0,0.8,0.6,0.45,0.3], optimizer=optimizer, metrics=['mean_squared_error'])
model.fit(
        dataset_train,
        epochs=30,
        validation_data=dataset_validation,
        callbacks=[
            tf.keras.callbacks.LearningRateScheduler(lambda i: 0.0005 * 10 ** -(i/10)),
            keras.callbacks.TensorBoard(log_dir=tensorboard_path, profile_batch=0),
        ],
    )
```

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by phemmer** [comment]
After finally figuring out how to get TF2 to stay in eager mode, allowing me to attach a debugger, I finally found where this unknown loss is coming from.
I have a kernel regularizer (L1L2) on one of the layers in the model, and as per [the documentation](https://www.tensorflow.org/api_docs/python/tf/keras/regularizers/Regularizer):

> These penalties are summed into the loss function that the network optimizes.

And if we look at the value:
```python
>>> model.layers[4].losses
[<tf.Tensor: shape=(), dtype=float32, numpy=2942.604>]
```

So I think that answers the issue. It is documented, just not on the model documentation page.

It still seems a little odd to me that the regularization loss is so extremely high. If this is expected, then I guess this issue can be closed out.
